cheatsheets/Input_Validation_Cheat_Sheet.md

```diff
@@ -166,42 +166,62 @@ The upload feature should be using a whitelist approach to only allow specific f
 
 # Email Address Validation
 
-## Email Validation Basics
+## Syntactic Validation
 
-Many web applications do not treat email addresses correctly due to common misconceptions about what constitutes a valid address. Specifically, it is completely valid to have an mailbox address which:
+The format of email addresses is defined by [RFC 5321](https://tools.ietf.org/html/rfc5321#section-4.1.2), and if far more complicated than most people realise. As an example, the following are all considered to be valid email addresses:
 
-- Is case sensitive in the local portion of the address (left of the rightmost `@` character).
-- Has non-alphanumeric characters in the local-part (including `+` and `@`).
-- Has zero or more labels.
+- `"><script>alert(1);</script>"@example.org`
+- `user+subaddress@example.org`
+- `user@[IPv6:2001:db8::1]`
+- `" "@example.org`
 
-At the time of writing, [RFC 5321](https://tools.ietf.org/html/rfc5321) is the current standard defining SMTP and what constitutes a valid mailbox address. Please note, email addresses should be considered to be public data.
+Properly parsing email addresses for validity with regular expressions is very complicated, although there are a number of [publicly documents regex](https://tools.ietf.org/id/draft-seantek-mail-regexen-03.html#rfc.section.3).
 
-Many web applications contain computationally expensive and inaccurate regular expressions that attempt to validate email addresses. Recent changes to the landscape mean that the number of false-negatives will increase, particularly due to:
+The biggest caveat on this is that although the RFC defines a very flexible format for email addresses, most real world implementations (such as mail servers) use a far more restricted address format, meaning that they will reject addresses that are *technically* valid.  Although they may be technically correct, these addresses are of little use if your application will not be able to actually send emails to them.
 
-- Increased popularity of sub-addressing by providers such as Gmail (commonly using `+` as a token in the local-part to affect delivery)
--  New [gTLDs](https://en.wikipedia.org/wiki/Generic_top-level_domain) with long names (many regular expressions check the number and length of each label in the domain)
+As such, the best way to validate email addresses is to perform some basic initial validation, and then pass the address to the mail server and catch the exception if it rejects it. This means that any the application can be confident that its mail server can send emails to any addresses it accepts. The initial validation could be as simple as:
 
-Following [RFC 5321](https://tools.ietf.org/html/rfc5321), best practice for validating an email address would be to:
+- The email address contains two parts, separated with an `@` symbol.
+- The email address does not contain dangerous characters (such as backticks, single or double quotes, or null bytes
```

Comment:
Close parentheses. I'd say dangerous characters should be defined based on how the email is used. It needs to be context aware. You think we should add that point?

Reply:
Added a note about this in 18a878d - is that what you were thinking?